joomla/joomla-cms
# [5.1] Add exceptions to deleted files check for JS files to be kept on update

Pull Request for Issue # .

### Summary of Changes

This pull request (PR) adds 4 folders to the list of folders not to be checked in the `build/deleted_file_check.php` script which is used to check if there are deleted files or folders to be added to the `administrator/components/com_admin/script.php` file.

The added folders belong to javascript media files which were removed from the sources but shall not be removed when updating the CMS for b/c reasons - there might be layout overrides still using them which will break when the media files were removed.

The `/media/legacy/js` folder was added due to PR #42776 , which has removed some of the javascript files in that folder. There are still files left in that folder in the source code, but for these the same b/c policy will apply when they would be removed, so the complete folder is added. See also my current PR #42935 .

The other folders belong to changing modals to dialogues (PRs #42082 and #42423 ). See my closed PR #42567 for details why these folders need to be added.

### Testing Instructions

The tests require PHP CLI so you can run PHP scripts on a command line (Linux shell or Windows CMD, no need for WSL) and a local Git clone of the 5.1-dev branch so the `build` folder is available. They can be executed on a clean branch without the need to run `composer install` or `npm install`.

The instructions below are made with Linux file paths, i.e. separator `/`. When being on Windows, just replace that by a backslash `\`.

1. Download the 5.1.0-alpha2 and the latest 5.1 nightly build installation packages:
- https://github.com/joomla/joomla-cms/releases/download/5.1.0-alpha2/Joomla_5.1.0-alpha2-Alpha-Full_Package.zip
- https://developer.joomla.org/nightlies/Joomla_5.1.0-beta1-dev-Development-Full_Package.zip
2. Unzip the packages into separate folders, e.g. when using the `tmp` folder of your Git clone and the package file names as folder names:
- ./tmp/Joomla_5.1.0-alpha2-Alpha-Full_Package
- ./tmp/Joomla_5.1.0-beta1-dev-Development-Full_Package
3. Open a command window (Linux shell or Windows CMD) and change directory to the root folder of your Git clone.
4. Run the following command to check the two folders from step 2 for deleted files and folders:
```
php ./build/deleted_file_check.php --from=./tmp/Joomla_5.1.0-alpha2-Alpha-Full_Package --to=./tmp/Joomla_5.1.0-beta1-dev-Development-Full_Package
```
5. Check the 2 created text files `./build/deleted_files.txt` and `./build/deleted_folders.txt`.
Result: See section "Actual result BEFORE applying this Pull Request" below.
6. Save the 2 text files from step 4 for later comparison.
7. Apply the changes from this PR.
8. Repeat steps 3 and 4 and compare the 2 text files with those saved in step 5.
Result: See section "Expected result AFTER applying this Pull Request" below.

### Actual result BEFORE applying this Pull Request

File `./build/deleted_files.txt` includes the following files:
```
'/media/com_cpanel/js/admin-add_module.js',
'/media/com_cpanel/js/admin-add_module.min.js',
'/media/com_cpanel/js/admin-add_module.min.js.gz',
'/media/com_modules/js/admin-module-edit.js',
'/media/com_modules/js/admin-module-edit.min.js',
'/media/com_modules/js/admin-module-edit.min.js.gz',
'/media/legacy/js/treeselectmenu.js'
'/media/legacy/js/treeselectmenu.min.js'
'/media/legacy/js/treeselectmenu.min.js.gz'
'/media/mod_multilangstatus/js/admin-multilangstatus.js',
'/media/mod_multilangstatus/js/admin-multilangstatus.min.js',
'/media/mod_multilangstatus/js/admin-multilangstatus.min.js.gz',
```

File `./build/deleted_folders.txt` includes the following folders:
```
'/media/mod_multilangstatus/js',
'/media/mod_multilangstatus',
```

### Expected result AFTER applying this Pull Request

Files `./build/deleted_files.txt` and `./build/deleted_folders.txt` do NOT include the files and folders mentioned in the previous section "Actual result BEFORE applying this Pull Request".

### Link to documentations
Please select:
